pkg/ipamd/ipamd.go

```diff
@@ -502,8 +508,10 @@ func (c *IPAMContext) getLocalPodsWithRetry() ([]*k8sapi.K8SPodInfo, error) {
 func (c *IPAMContext) StartNodeIPPoolManager() {
 	sleepDuration := ipPoolMonitorInterval / 2
 	for {
-		time.Sleep(sleepDuration)
-		c.updateIPPoolIfRequired()
+		if !c.disableENIProvisioning{
+			time.Sleep(sleepDuration)
+			c.updateIPPoolIfRequired()
+		}
```

Comment:
If the `time.Sleep()` call is inside the if-statement, the `nodeIPPoolReconcile()` will be called twice as often. It's a bit weird, but since that should not trigger EC2 API calls we should be fine.

Reply:
hmmm, that is a good point. Let me make a compensate on the interval in the whitelist code change. Thanks!